instrumentation/aws_lambda/lib/opentelemetry/instrumentation/aws_lambda/wrap.rb

```diff
@@ -43,7 +43,7 @@ def wrap_lambda(event:, context:, handler:, flush_timeout: DEFAULT_FLUSH_TIMEOUT
               begin
                 response = yield
 
-                unless span.attributes.key?(OpenTelemetry::SemanticConventions::Trace::HTTP_STATUS_CODE)
+                if span.respond_to?(:attributes) && !span.attributes.key?(OpenTelemetry::SemanticConventions::Trace::HTTP_STATUS_CODE)
```

Comment:
Probably can just check if the span is recording, right?
```suggestion
                if span.recording? && !span.attributes.key?(OpenTelemetry::SemanticConventions::Trace::HTTP_STATUS_CODE)
```